app/controllers/transport/border/active/IdentificationNumberController.scala

```diff
@@ -87,4 +105,32 @@ class IdentificationNumberController @Inject() (
       _              <- sessionRepository.set(updatedAnswers)
     } yield Redirect(navigator.nextPage(IdentificationNumberPage(activeIndex), updatedAnswers, departureId, mode))
 
+  private def identificationPageIe170(activeIndex: Index)(implicit request: MandatoryDataRequest[_]): Option[Identification] =
+    request.userAnswers.get(IdentificationPage(activeIndex))
+
+  private def identificationPageIe015(activeIndex: Index)(implicit request: MandatoryDataRequest[_]): Option[String] =
+    request.userAnswers.departureData.Consignment.ActiveBorderTransportMeans.flatMap(
+      x => x.lift(activeIndex.position).flatMap(_.typeOfIdentification)
+    )
+
+  private def getReferenceDataFor15(activeIndex: Index)(implicit request: MandatoryDataRequest[_]): Option[Future[Identification]] =
+    identificationPageIe015(activeIndex).map {
+      aaa =>
+        service.getBorderMeansIdentification(aaa)(hc(request.request))
```

Comment:
Do you need to be using `MandatoryDataRequest`? If you change these to `DataRequest` you can probably get rid of `(hc(request.request))`